### ISSUE TITLE
Oneshot Optimization

### ISSUE THREAD

**Issue by thsubaku9** [thsubaku9/Neural_Nets_Stuff#1]
Currently, we need to iterate through 100s of epochs before achieving true Oneshot capabilities. Hyperparameter tuning, optimizer type, additional loss, etc. need to be improved

**Comment by thsubaku9** [comment]
vanilla GradientDescent might work for Oneshot. AdaGrad gets stuck in local min, RMS overshoots. Currently using Adam but it's causing varying results. Maybe a gradual change in learning rate and beta params might help?

**Comment by thsubaku9** [comment]
*Siamese networks will work but that isn't what we're searching for. Those things exist for the sake of closeness approximation scoring.
